### PR TITLE
Apply some constraints to genome creation

### PR DESCRIPTION
**Pull request by ghost**
In this system the cost of evaluation will be high, so we should ensure that the same upgrade sequence is not evaluated more than once. There should also be one genome which tries going straight to the goal.

This could save a lot of compile time.

### CODE REVIEW

File: src/population.c [hunk 1]
```diff
@@ -102,6 +130,37 @@ void population_free(sc_population * population)
...
+        genome_array = population->individual;
+
+    for (i = 0; i < genome_index; i++) {
+        if (genome_array[i]->steps != genome->steps)
+            continue;
+        if (memcmp((void*)&genome_array[i]->change,
+                   genome->change, genome->steps*sizeof(sc_system_state)) == 0)
+            return (1 == 0);
+    }
+    return (1 == 1);
```
Comment (leeming) [inline src/population.c]: Confused why you don't just return 1? Is there something else happening here?

File: src/population.c [hunk 2]
```diff
@@ -277,19 +336,33 @@ int population_next_generation(sc_population * population)
...
+                                  population_parent(population),
+                                  population_parent(population),
+                                  population->next_generation[i]);
+            if (retval != 0)
+                return 30 + retval;
+
+            /* if there is repeated failure to create a unique
+               child genome */
+            tries++;
+            if (tries > 1000)
```
Comment (leeming) [inline src/population.c]: SC_MAX_TRIES 1000 ?

Comment (ghost) [inline src/population.c]: 1000 seemed like a number big enough to have a chance of uniqueness.

Comment (ghost) [inline src/population.c]: But I suppose it could be a define

**Pull request merged**